cmake/nlohmann-json.cmake

```diff
@@ -33,6 +33,8 @@ find_package(nlohmann_json QUIET)
 set(nlohmann_json_clone FALSE)
 if(nlohmann_json_FOUND)
   message(STATUS "nlohmann::json dependency satisfied by: package")
+elseif(TARGET nlohmann_json)
```

Comment:
Wondering the scenario that `nlohmann_json_FOUND` is not set, but `nlohann_json` target is available.

Reply:
`nlohmann_json_FOUND` true only when the library installed on the system like package, not as a submodule or third party usage in another project. 

In my use case, we need this library for another purpose as a third party, so CMake acts that `nlohmann_json` is a CMake target which is the correct in project perspective. If we compile `opentelemetry-cpp + my project + my nlohmann_json` together, `opentelemetry-cpp` will use my `nlohmann_json` library because it is already known as a CMake target by CMake.

Otherwise, it throws an error which I already explained in the bug description.

Reply:
nlohmann_json may be imported by `add_subdirectory`.

> Wondering the scenario that `nlohmann_json_FOUND` is not set, but `nlohann_json` target is available.